Faildump mapping

aspark21 opened the pull request:
I know there's the web browser approach to loading the faildumps but we're using moodle-docker within our CI pipeline so looking to retrieve the files in bulk without opening a browser. Just another option. Found an old commit from Andrew that had started this and updated this to align with the path which has since been defined for faildumps.

Tested in our pipeline and works well.

Github Actions will have the usual failures because #195 not yet in but this passes here - https://github.com/aspark21/moodle-docker/commits/merge-them-all / https://github.com/aspark21/moodle-docker/actions/runs/1535306113 - where I've merged 195, this and the mysql8 change(171) into one branch.

scara commented on bin/moodle-docker-compose:
```diff
@@ -128,6 +128,11 @@ then
     dockercompose="${dockercompose} -f ${basedir}/phpunit-external-services.yml"
 fi
 
+if [[ ! -z "$MOODLE_DOCKER_BEHAT_FAILDUMP" ]];
```
Hi @aspark21,
what about testing even if it is actually a directory too?
Kind of `[[ -d "$MOODLE_DOCKER_BEHAT_FAILDUMP" ]]`.

HTH,
Matteo

aspark21 replied:
Fair point updated accordingly

scara commented on bin/moodle-docker-compose.cmd:
```diff
@@ -64,6 +64,10 @@ IF NOT "%MOODLE_DOCKER_PHPUNIT_EXTERNAL_SERVICES%"=="" (
     SET DOCKERCOMPOSE=%DOCKERCOMPOSE% -f "%BASEDIR%\phpunit-external-services.yml"
 )
 
+IF NOT "%MOODLE_DOCKER_BEHAT_FAILDUMP%"=="" (
```
Hi @aspark21,
same as above, what about adding a dir check even here?
Kind of `exist "%MOODLE_DOCKER_BEHAT_FAILDUMP%\nul"`.

HTH,
Matteo

aspark21 replied:
Aligned the checks with those done for MOODLE_DOCKER_WWWROOT but nested as this is not a mandatory env variable

aspark21 commented:
re-based on master now that #195 integrated so all tests are passing 😄 

 I believe I've addressed all of your comments @scara 

This has dual benefit of allowing access to faildumps but also to use a timing file when running test suites - https://github.com/moodlehq/moodle-docker/issues/84#issuecomment-991993273 (that might warrant some documentation / tweak to config.docker-template.php in a separate pull request)

stronk7 commented:
Here we go, thanks @aspark21 ! 😎
